app/src/main/java/net/gsantner/markor/activity/MainActivity.java

```diff
@@ -182,10 +192,8 @@ private void optShowRate() {
 
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-        MarkorPermissionChecker permc = new MarkorPermissionChecker(this);
-        permc.checkPermissionResult(requestCode, permissions, grantResults);
-
-        if (_cu.checkExternalStoragePermission(this, false)) {
+        _permc.checkPermissionResult(requestCode, permissions, grantResults);
```

Comment:
Made the Checker a member

Reply:
>         _permc = new MarkorPermissionChecker(this);

Problem: This keeps a reference to the activity as a member.   
Suggestion:The better option here is probably similar to ContextUtils, to pass the activity reference on call as argument. Whats not stored somewhere hardly can make a memory leak 😄